todo_app/app.py

```diff
@@ -24,47 +24,50 @@ def create_app():
     login_manager.login_manager.init_app(app)
     
     # All the routes and setup code etc
-    @app.route('/', methods=['GET'])
+    @app.route('/')
     @login_required
     def index():
-        if app.config['LOGIN_DISABLED'] == "True":
+        if app.config["LOGIN_DISABLED"] == "True":
             user_role = "writer"
         else:
             user_role = current_user.role
         my_statuses = app.config['STATUSES']
-        mongo_todo_list = view_model(get_mongo_todo_items(),get_mongo_todo_statuses(), my_statuses)
+        todo_list = view_model(get_todo_items(),get_todo_statuses(), my_statuses)
         app.logger.info('Processing get cards request')
-        return render_template('index.html', view_model_items=mongo_todo_list, view_role=user_role)
+        return render_template('index.html', view_model_items=todo_list, view_role=user_role)
 
 
     @app.route('/create', methods=['POST'])
     @login_required
     def new_todo():
-        if current_user.role == "writer":
+        if app.config["LOGIN_DISABLED"] == "True" or current_user.role == "writer":
             last_update = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.%fZ')
-            mongo_post(request.form['add_todo'], request.form['add_desc'], request.form['due_date'], last_update)
+            database_post(request.form['add_todo'], request.form['add_desc'], request.form['due_date'], last_update)
             app.logger.info('Processing create new card request')
-            return redirect('/')
+            return index()
```

Comment:
Why did you change this?
`return redirect('/')` is better than `return index()`

It will update the browser's URL correctly, and the user can hit refresh without resubmitting the form.

And by getting the browser to make a fresh request, you will avoid potential problems if the index function tries to look at the current request.

Reply:
I'll change it back, I was trying to see if it helped with the redirect issue I was having during the exercise.  It didn't but everything seemed to work the same so I left the update in.